### PR TITLE
fixed issue #48026: unintended mask shape

### PR DESCRIPTION
**Pull request by SebastianThomas1**

### CODE REVIEW

File: tensorflow/python/keras/layers/core.py [hunk 1]
```diff
@@ -121,14 +121,16 @@ def __init__(self, mask_value=0., **kwargs):
...
   def compute_mask(self, inputs, mask=None):
-    return K.any(math_ops.not_equal(inputs, self.mask_value), axis=-1)
+    return K.any(math_ops.not_equal(inputs, self.mask_value),
+                 axis=list(range(2, len(inputs.shape))))
 
   def call(self, inputs):
+    axes = list(range(2, len(inputs.shape)))
     boolean_mask = K.any(
-        math_ops.not_equal(inputs, self.mask_value), axis=-1, keepdims=True)
+        math_ops.not_equal(inputs, self.mask_value), axis=axes, keepdims=True)
```
Comment (fchollet) [inline tensorflow/python/keras/layers/core.py:130]: Typically, a mask is used to mask timesteps in a sequence (3D data). You want to get a 2D mask of shape (batch_size, length) where `mask[s, t]` is a boolean marking whether timestep `t` of sample `s` is masked or not. We should only reduce the last (features) axis.

Comment (SebastianThomas1) [reply tensorflow/python/keras/layers/core.py:130]: Yes, I have 4D data of shape `(batch_size, n_time_steps, n_features, n_channels)`, as explained in the issue #48026. And with the current version of tf.keras.layers.Masking, those data is not handled correctly.

I do not see why we should only reduce the last (features) axis since features may be multidimensional. My bug fix still handles the 3D case correctly (in this case, `axes` will be `[2]`, so `axis=axes` will have the same effect as `axis=-1`, doesn't it? Are there cases with data of higher dimension, or other cases, that are not correctly handled with my fix?
